azurerm/internal/provider/provider.go

```diff
@@ -228,6 +228,13 @@ func azureProvider(supportLegacyTestSuite bool) terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("ARM_STORAGE_USE_AZUREAD", false),
 				Description: "Should the AzureRM Provider use AzureAD to access the Storage Data Plane API's?",
 			},
+
+			"custom_correlation_request_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     schema.EnvDefaultFunc("ARM_CORRELATION_REQUEST_ID", ""),
+				Description: "The value of the x-ms-correlation-request-id header (otherwise an auto-generated UUID will be used).",
+			},
```

Comment:
we can remove this and instead use `os.GetEnv` below since this isn't an intentionally user exposed value